scheduler/server/load_based_sched_alg.go

```diff
@@ -0,0 +1,723 @@
+
+		lbs.getTaskAllocations(totalWorkers - totalTasks)
+	}
+
+	stopTasks := lbs.buildTaskStopList()
+
+	return stopTasks
+}
+
+// getCurrentPctsSpread is used to measure how well the current worker assignment matches the target loads.
+// It computes each class's difference between the target load pct and actual load pct.  The 'spread' value
+// is the difference between the min and max differences across the classes.  Eg: if 30% was the load target
+// for class A and class A is using 50% of the workers, A's pct difference is -20%.  If class B has a target
+// of 15% and is only using 5% of workers, B's pct difference is 10%.  If A and B are the only classes, the
+// pctSpread is 25% (5% - -20%).
+func (lbs *LoadBasedAlg) getCurrentPctsSpread(totalWorkers int) int {
+	if len(lbs.jobClasses) < 2 {
+		return 0
+	}
+	minPct := 0
+	maxPct := 0
+	for _, className := range lbs.classByDescLoadPct {
+		jc := lbs.jobClasses[className]
+		currPct := int(math.Floor(float64(jc.origNumRunningTasks) * 100.0 / float64(totalWorkers)))
+		pctDiff := jc.origTargetLoadPct - currPct
+		minPct = min(minPct, pctDiff)
+		maxPct = max(maxPct, pctDiff)
+	}
+
+	return maxPct - minPct
+}
+
+// GetClassByDescLoadPct get a copy of the config's class by descending load pcts
+func (lbs *LoadBasedAlg) GetClassByDescLoadPct() []string {
+	lbs.config.classLoadPctsMu.RLock()
+	defer lbs.config.classLoadPctsMu.RUnlock()
+	copy := []string{}
+	for _, v := range lbs.config.classByDescLoadPct {
+		copy = append(copy, v)
+	}
+	return copy
+}
+
+// GetClassLoadPcts return a copy of the ClassLoadPcts converting to int32
+func (lbs *LoadBasedAlg) GetClassLoadPcts() map[string]int32 {
+	lbs.config.classLoadPctsMu.RLock()
+	defer lbs.config.classLoadPctsMu.RUnlock()
+	copy := map[string]int32{}
+	for k, v := range lbs.config.classLoadPcts {
+		copy[k] = int32(v)
+	}
+	return copy
+}
+
+// LocalCopyClassLoadPcts return a copy of the ClassLoadPcts leaving as int
+func (lbs *LoadBasedAlg) LocalCopyClassLoadPcts() map[string]int {
+	lbs.config.classLoadPctsMu.RLock()
+	defer lbs.config.classLoadPctsMu.RUnlock()
+	copy := map[string]int{}
+	for k, v := range lbs.config.classLoadPcts {
+		copy[k] = v
+	}
+	return copy
+}
+
+// SetClassLoadPcts set the scheduler's class load pcts with a copy of the input class load pcts
+func (lbs *LoadBasedAlg) SetClassLoadPcts(classLoadPcts map[string]int32) {
+	lbs.config.classLoadPctsMu.Lock()
+	defer lbs.config.classLoadPctsMu.Unlock()
+
+	// build a list that orders the classes by descending pct.
+	keys := []string{}
+	for key := range classLoadPcts {
+		keys = append(keys, key)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return classLoadPcts[keys[i]] > classLoadPcts[keys[j]]
+	})
+	lbs.config.classByDescLoadPct = keys
+
+	// set the load pcts - normalizing them if the don't add up to 100
+	lbs.config.classLoadPcts = map[string]int{}
+	pctTotal := 0
+	for _, val := range classLoadPcts {
+		pctTotal += int(val)
+	}
+	if pctTotal != 100 {
```

Comment:
Why not just reject this? (see also other comment about having a "reserve" bucket/class)